model/supervised_finetuning/utils.py

```diff
@@ -32,8 +36,51 @@ def get_tokenizer(conf):
     return tokenizer
 
 
+# placeholder for now
+def preprocess_qa(eval_pred):
+    return (eval_pred.predictions, eval_pred.label_ids)
+
+
+def postprocess_summarization(preds, labels):
+    preds = [pred.strip() for pred in preds]
+    labels = [label.strip() for label in labels]
+
+    preds = ["\n".join(nltk.sent_tokenize(pred)) for pred in preds]
+    labels = ["\n".join(nltk.sent_tokenize(label)) for label in labels]
+
+    return preds, labels
+
+
+def preprocess_summarization(eval_pred, tokenizer, ignore_pad_token_for_loss=True):
+    preds, labels = eval_pred
+    decoded_preds = tokenizer.batch_decode(preds, skip_special_tokens=True)
+    if ignore_pad_token_for_loss:
+        labels = np.where(labels != -100, labels, tokenizer.pad_token_id)
+    decoded_labels = tokenizer.batch_decode(labels, skip_special_tokens=True)
+
+    decoded_preds, decoded_labels = postprocess_summarization(decoded_preds, decoded_labels)
+    return decoded_preds, decoded_labels
+
+
+def get_metrics(conf, tokenizer):
+    qa_datasets = ["squad_v2", "adversarial_qa", "trivia_qa_context", "trivia_qa_noconext"]
+    summarization_datasets = ["xsum", "cnn_dailymail", "samsum", "multi_news"]
+
+    # the reason behind using a list is that we might want to extend the list of our
+    # metrics in the future for more thorough evaluation
+    if any(dataset in qa_datasets for dataset in conf.datasets):
+        metrics, preprocess_fn = [evaluate.load("squad_v2")], preprocess_qa
+    elif any(dataset in summarization_datasets for dataset in conf.datasets):
+        metrics, preprocess_fn = [evaluate.load("rouge")], partial(
+            preprocess_summarization, tokenizer, ignore_pad_token_for_loss=conf.ignore_pad_token_for_loss
```

Comment:
Is ignore_pad_token_for_loss defined in the config?

Reply:
> Is ignore_pad_token_for_loss defined in the config?

It is one of the default arguments of huggingface trainer, so I thought it would be in the config.